src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/delete_interaction_choice_set_request.cc

```diff
@@ -157,12 +182,48 @@ void DeleteInteractionChoiceSetRequest::SendVrDeleteCommand(
   msg_params[strings::type] = hmi_apis::Common_VRCommandType::Choice;
   msg_params[strings::grammar_id] = (*choice_set)[strings::grammar_id];
   choice_set = &((*choice_set)[strings::choice_set]);
-  for (uint32_t i = 0; i < (*choice_set).length(); ++i) {
-    msg_params[strings::cmd_id] = (*choice_set)[i][strings::choice_id];
-    SendHMIRequest(hmi_apis::FunctionID::VR_DeleteCommand, &msg_params);
+  {
+    sync_primitives::AutoLock auto_lock(requests_lock_);
+    for (uint32_t i = 0; i < (*choice_set).length(); ++i) {
+      msg_params[strings::cmd_id] = (*choice_set)[i][strings::choice_id];
+      const uint32_t delte_cmd_hmi_corr_id = SendHMIRequest(
+          hmi_apis::FunctionID::VR_DeleteCommand, &msg_params, true);
+      sent_requests_.insert(delte_cmd_hmi_corr_id);
+    }
   }
 }
 
+void DeleteInteractionChoiceSetRequest::
+    SendDeleteInteractionChoiceSetResponse() {
+  const mobile_apis::Result::eType result_code =
+      MessageHelper::HMIToMobileResult(response_result_code_);
+  response_result_ =
```

Comment:
this compare should be `IsHMIResultSuccess(response_result_code_)`, perhaps this should be done on every response like so (all responses should be considered):

in on_event:
response_result_ = response_result_ && IsHMIResultSuccess